explainer.md

```diff
@@ -18,96 +18,107 @@ permanent "block" policies, or
+
+1.  **Insufficiency of existing mitigations** The present day permissions spam
+    and abuse mitigation approach has an architectural upper bound on user
+    protection because the model relies on the website to choose when to trigger
+    the permission request prompt rather than capturing a reliable signal of
+    user intent. Requiring a user gesture for the Permission API (or similar)
+    does not solve this problem as there are many ways of tricking a user into
+    providing such a gesture.
+
+1.  **Context**: Ideally, a site's developer will request access as part of a
+    contextual flow that helps users understand what's being asked for and why,
+    enabling quick and confident responses. Often, however, permission requests
+    are correlated poorly with user expectations, up to and including prompts
+    that can come out of nowhere (see example 1). This places a burden on user
+    agents' presentation of the request. The user agent has no semantic
+    understanding of events taking place in the content area prior to the
+    permission request. User agents could make better decisions and provide
+    better prompts if they could make well-founded assumptions about the nature
+    of the user's interaction in the content area, and the user's intent.
+
+    ![](images/image1.png) \
+    *Example 1. A notification permission prompt on a news site (contents
+    blurred), shown after the user has clicked on the empty area next to the
+    article content. The user finds this prompt interruptive as they had no
+    interest in subscribing to notifications, and they will likely struggle to
+    understand why the prompt was shown to begin with.*
+
+1.  **Location**: In the ideal case above, users will interact with something on
+    a site that triggers a prompt. In less ideal cases, the user might not have
+    interacted with anything at all, or they may have interacted with an element
+    that was unrelated to the request. Given this uncertainty, user agents rely
+    on common placement of the permission prompt, usually in the top-left of the
+    page. Even in the best case, this has the unfortunate effect of shifting the
+    point to which users need to pay attention from the thing they clicked on to
+    some distant part of the user agent's UI (see example 2). User agents could
+    make better decisions and provide better prompts if they could make
+    well-founded assumptions about the nature of the user's interaction in the
+    content area, and the user's current area of focus. In effect, we think
+    there is a benefit to semantic markup for permissions.
+
+    ![](images/image2.png) \
+    *Example 2. An example where the permission prompt is far away from the
+    user's current area of focus. The permission prompt was triggered because
+    the user has just clicked on the crosshair icon in the bottom right, but the
+    prompt is easy to miss since it's on the opposite side of the page.*
```

Comment:
Again, this is not a universal issue. For example, on iOS this wouldn't be an issue because the prompt is modal. 

I think we have to be careful to not design features to overcome UX limitations that could be addressed in other ways. For instance, there is nothing stopping browsers from not doing the door hanger model for permissions.  
  
![Screenshot 2023-12-20 at 11 45 38](https://github.com/WICG/PEPC/assets/870154/0e2e05ab-187b-4b77-a716-ed1418754659)

In fact, notice that Safari doesn't do the door hanger for the Notifications API. It does the modal prompt on both macOs and iOS.

Reply:
Extracted into https://github.com/WICG/PEPC/issues/8